### PR TITLE
Remove depricated type_traits usage.

### PR DESCRIPTION
**Pull request by jzmaddock**

### CODE REVIEW

**Comment by zyv** [comment]
Duplicated by #24.

**Comment by Bklyn** [comment]
It'd be great to get this merged before the next official Boost release (1.62)!

**Comment by greglandrum** [comment]
Agreed, having either this or #24 merged would really be excellent.

File: include/boost/iostreams/detail/is_dereferenceable.hpp [hunk 1]
```diff
@@ -9,9 +9,8 @@
 #ifndef BOOST_IOSTREAMS_DETAIL_IS_DEREFERENCEABLE_HPP_INCLUDED
 #define BOOST_IOSTREAMS_DETAIL_IS_DEREFERENCEABLE_HPP_INCLUDED
 
-# include <boost/type_traits/detail/bool_trait_def.hpp>
-# include <boost/type_traits/detail/template_arity_spec.hpp>
 # include <boost/type_traits/remove_cv.hpp>
+# include <boost/type_traits/integral_constant.hpp>
 # include <boost/mpl/aux_/lambda_support.hpp>
```
Comment (Flast) [inline include/boost/iostreams/detail/is_dereferenceable.hpp]: It seems this header is also no longer used.

**Pull request closed without merging**